Lib/email/message.py

```diff
@@ -1041,7 +1041,10 @@ def iter_attachments(self):
         maintype, subtype = self.get_content_type().split('/')
         if maintype != 'multipart' or subtype == 'alternative':
             return
-        parts = self.get_payload().copy()
+        try:
+            parts = self.get_payload().copy()
+        except AttributeError:
+            raise TypeError('Expected payload as list, got %s' % type(self._payload))
```

Comment:
Use `raise TypeError('Expected payload as list, got %s' % type(self._payload)) from None` to hide the `AttributeError` from the stack trace.